tidepods/write_pfs.py

```diff
@@ -1,86 +1,92 @@
+from string import Template
+
 import fiona
+from fiona.transform import transform_geom
 import numpy as np
 from shapely.geometry import shape, Point
-from fiona.transform import transform_geom
-from string import Template
+
 
 def transform_shape(infile):
-    """Reprojects polygon to EPSG:4326 and return as shapely object
+    '''Reprojects polygon to EPSG:4326 and return as shapely object
 
     Parameters
     ----------
-    infile: str
-        path to polygon
+    infile : str
+            File path to polygon.
 
     Returns
     -------
-    shapely object : polygon
-    """
+    shp : shapely object
+
+    '''
     with fiona.open(infile) as c:
         infile_4326 = transform_geom(c.crs.get("init"), 'epsg:4326', c[0]['geometry'])
         shp = shape(infile_4326)
-    return (shp)
+    return shp
+
 
 def generate_pts(infile):
-    """Generates fixed distance points within a polygon
+    '''Generates fixed distance points within a polygon
 
     Parameters
     ----------
-    infile: str
-        path to polygon
+    infile : str
+            File path to polygon.
 
     Returns
     -------
-    list : list of shapely points for points within infile
-    """
+    plist : list
+            List of shapely points for points within infile
+
+    '''
     shp = transform_shape(infile)
     bounds = shp.bounds
     minx, miny, maxx, maxy = bounds
-    plist=[]
+    plist = []
     for x in np.arange(minx, maxx, 0.125):
         for y in np.arange(miny, maxy, 0.125):
-            p = (Point(x,y))
+            p = (Point(x, y))
             if p.within(shp):
                 plist.append(p)
-    return (plist)
+    return plist
+
 
 def write_pfs(shp, template, date, pfs):
-    """Creates pfs file for input to MIKE
+    '''Creates pfs file for input to MIKE
 
     Parameters
     ----------
-    shp: str
-        path to input shapefile
-    
-    template: str
-        path to template textfile
-
-    date: str
-        image acquisiton date (YYYYMMDD)
-        
-    pfs: str
-        path to pfs file
+    shp : str
+          File path to polygon.
 
-    Returns
-    -------
-    pfs file: a .pfs file for input to MIKE
-    """
+    template : str
+               File path to template textfile.
+
+    date : datetime
+           Image acquisiton date (YYYYMMDD).
+
+    pfs : str
+          File path to pfs file.
+
+    '''
     date_str = str(date)
     acq_year = date_str[:4]
     plist = generate_pts(shp)
     formatted_pts = []
     pid = 1
+    newline = '\n'
     for p in plist:
-        formatted_pts.append(r'[Point_' + str(pid) + ']\n' +
-                        r'description = ' + str(pid) + '\n' +
-                        r'y = ' + str(p.coords[0][0]) + '\n' +
-                        r'x = ' + str(p.coords[0][1]) + '\n' +
-                        r'EndSect // Point_' + str(pid) + '\n'
-            )
-        pid +=1
-    
+        point_text = ''.join(['[Point_', str(pid), ']', newline,
```

Comment:
A cleaner solution here would to make a formatting string like

```python
POINT_FMT = (
'''[Point_{pid}]
description = {pid}
y = {coords[0]}
x = {coords[1]}
EndSect // Point_{pid}''')
```
and then create your strings with 

```python
points_str = POINT_FMT.format(pid=pid, coords=p.coords[0])
```

Reply:
You can also drop the brackets and do this:

```
POINT_FMT = """
[Point_{pid}]
description = {pid}
y = {coords[0]}
x = {coords[1]}
EndSect // Point_{pid}
""".strip()
```

Matter of taste I guess :)